src/main/scala/Main.scala

```diff
@@ -39,50 +49,60 @@ object Main
 
     val numList = OGnumList.reverse
 
-    val list = List(
-      ("thousand", "thousand"),
-      ("million", "million"),
-      ("billion", "milliard"),
-      ("trillion", "billion"),
-      ("quadrillion", "billiard"),
-      ("quintillion", "trillion"),
-      ("sextillion", "trilliard"))
-
-    val numLoop = math.floor(numLength / 3).toLong
-
-    buildOut(numLoop, numLength, numList, list, "", "", 0)
+    val longList = List(
+      "thousand",
+      "million",
+      "milliard",
+      "billion",
+      "billiard",
+      "trillion",
+      "trilliard"
+    )
+
+    val shortList = List(
+      "thousand",
+      "million",
+      "billion",
+      "trillion",
+      "quadrillion",
+      "quintillion",
+      "sextillion"
+    )
+
+    buildOut(numLength, numList, shortList, longList, "", "", 0)
 
   }
 
   def buildOut(
-                numLoop:Long,
                 numLength:Long,
                 numList:ListBuffer[Long],
-                list:List[(String, String)],
+                shortList:List[String],
+                longList:List[String],
                 outStr:String,
                 outStr2:String, z:Int): Unit = {
 
-    z == (numLoop + 1) match {
-      case true =>
+      val numLoop = math.floor(numLength / 3).toLong
+      z match {
+      case z if numLoop + 1 == z  =>
         // end the loop
         val finalOut = outStr.replaceAll("( )+", " ").trim()
         val finalOut2 = outStr2.replaceAll("( )+", " ").trim()
         println("\nShort Form: " + finalOut + "\nLong Form: " + finalOut2 + "\n\n- - - - -\n")
 
-      case false =>
+      case _ =>
         numLength >= (((z + 1) * 3) + 1) match {
           case true =>
 
             val theOut =
-              list(z)._1 + " " +
+              shortList(z) + " " +
                 (
                   numList((z * 3) + 2).toString +
```

Comment:
these numList((z * 3) + 2).toString and similar are called a couple of times, would look better if you worked it out into a helper function instead of having the duplicate code